### PR TITLE
Updated logic for Vm Repair Reset Nic cmd(ResourceNotFound)

### PR DESCRIPTION
**Pull request by pselugar**
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- Updated logic for Reset Nic "ResourceNotFound" bug.
- Reset Nic Script now correctly picks `VNet ResourceGroup` for fetching `IpAddress` instead of using `ResourceGroup of VM`.
- Tested logic on several Test VM's.
### Related command
` az vm repair reset-nic -RgName $rgname  -VMName $name --subscription $SubId --verbose`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

File: src/vm-repair/azext_vm_repair/custom.py [hunk 1]
```diff
@@ -573,14 +573,15 @@ def reset_nic(cmd, vm_name, resource_group_name, yes=False):
         subnet_id_tokens = subnet_id.split('/')
```
Comment (dkulkarni-ms) [inline src/vm-repair/azext_vm_repair/custom.py:573]: Check my previous PR to this file or branch. Version of reset-nic may also needs to be changed.

**Comment by yonzhan** [comment]
vm repair

File: src/vm-repair/azext_vm_repair/custom.py (hunk 1 above)
Comment (pselugar) [reply src/vm-repair/azext_vm_repair/custom.py:573]: Okay working on it

Comment (pselugar) [reply src/vm-repair/azext_vm_repair/custom.py:573]: Done

**Review by haagha — APPROVED** [review]
Approved.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**